src/gui/attributetable/qgsdualview.cpp

```diff
@@ -979,6 +981,46 @@ void QgsDualView::showViewHeaderMenu( QPoint point )
   connect( sort, &QAction::triggered, this, [ = ]() {modifySort();} );
   mHorizontalHeaderMenu->addAction( sort );
 
+  mConfig.update( mLayer->fields() );
+  // get layer field index from column name
+  const int fieldIndex = mLayer->fields().indexFromName( mConfig.columns().at( col ).name );
+  const int fieldOrigin  = mLayer->fields().fieldOrigin( fieldIndex );
```

Comment:
The other field-specific actions like resize column use this approach:

```
fieldIndex = mConfig.mapVisibleColumnToIndex( col );
```

Can that same approach be used here? (It makes me nervous straying from the already existing, tested path!)

Reply:
This is what I tried first but it maps to a column index and not the actual field index. However, this made me realize that I do have to use mapVisibleColumnToIndex to get the correct index if some fields are hidden.
`const int fieldIndex = mLayer->fields().indexFromName( mConfig.columns().at( mConfig.mapVisibleColumnToIndex( col ) ).name );`